### PR TITLE
[price-pusher] Refactor Price Pusher to support other chains 

### PR DESCRIPTION
**Pull request by 0xfirefist**
moved shouldUpdate to price-config as it is closely related to it
pusher changed to controller + moved the code to push updates into evm.ts
evm.ts contains everything related to evm. PriceGetter, PricePusher, Contract

one change i do think should have been in the separate PR in this file price_pusher/src/pyth-price-listener.ts

### CODE REVIEW

**Comment by 0xfirefist** [comment]
I will send other PRs for above changes!

File: price_pusher/src/pyth-price-listener.ts [hunk 1]
```diff
@@ -1,21 +1,15 @@
-import {
-  EvmPriceServiceConnection,
-  HexString,
-  PriceFeed,
-} from "@pythnetwork/pyth-evm-js";
+import { HexString, PriceFeed } from "@pythnetwork/pyth-evm-js";
```
Comment (ali-bahjati) [inline price_pusher/src/pyth-price-listener.ts:1]: You can remove this too.

File: price_pusher/src/price-config.ts [hunk 2]
```diff
@@ -47,3 +48,84 @@ export function readPriceConfigFile(path: string): PriceConfig[] {
     return priceConfig;
   });
 }
+
```
Comment (ali-bahjati) [inline price_pusher/src/price-config.ts:51]: Why did you move it here? I think maybe it's better to be on `controller.ts`

Comment (0xfirefist) [reply price_pusher/src/price-config.ts:51]: it is very closely related to price config. 
if config changes we will need to update the should Update logic too. 
controller doesn't need to know anything related to priceConfig. 
I am thinking in this direction.

File: price_pusher/src/pyth-price-listener.ts (hunk 1 above)
Comment (0xfirefist) [reply price_pusher/src/pyth-price-listener.ts:1]: will update it in other prs
